### PR TITLE
feat: integrate cellxgene-ontology-guide with cellxgene-schema CLI

### PR DESCRIPTION
**Pull request by nayib-jose-gloria**
## Reason for Change

- https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/gh/chanzuckerberg/cellxgene-ontology-guide/102

## Changes

- Replace OntologyChecker with OntologyParser
- Remove OntologyChecker
- Delete convert.py (already marked for deprecation and had OntologyChecker use-cases)

## Testing

- Test against datasets (need to come up with a plan for broader testing--migration?)

## Notes for Reviewer

### CODE REVIEW

**Review by Bento007 — APPROVED** [review]

**Comment by Bento007** [comment]
Looks good. A few questions and a test pypi index to remove.

File: cellxgene_schema_cli/cellxgene_schema/validate.py [hunk 1]
```diff
@@ -115,7 +116,7 @@ def _has_forbidden_curie_ancestor(
         """
         for ontology_name in forbidden_def:
             for ancestor in forbidden_def[ontology_name]:
-                if ONTOLOGY_CHECKER.is_descendent_of(ontology_name, term_id, ancestor):
+                if ancestor in ONTOLOGY_PARSER.get_term_ancestors(term_id):
```
[the remark this answers is not in the document]
Comment (nayib-jose-gloria) [reply cellxgene_schema_cli/cellxgene_schema/validate.py:119]: We parse it from the term_id in our ontology_parser, we don't take it as an input anymore

File: cellxgene_schema_cli/cellxgene_schema/validate.py [hunk 2]
```diff
@@ -142,15 +143,15 @@ def _validate_curie_ancestors(
...
 
-        for ontology_name, ancestors in allowed_ancestors.items():
+        for _, ancestors in allowed_ancestors.items():
             for ancestor in ancestors:
                 if inclusive and term_id == ancestor:
                     checks.append(True)
 
-                is_valid_term_id = ONTOLOGY_CHECKER.is_valid_term_id(ontology_name, term_id)
-                is_valid_ancestor_id = ONTOLOGY_CHECKER.is_valid_term_id(ontology_name, ancestor)
+                is_valid_term_id = ONTOLOGY_PARSER.is_valid_term_id(term_id)
```
[the remark this answers is not in the document]
Comment (nayib-jose-gloria) [reply cellxgene_schema_cli/cellxgene_schema/validate.py:151]: We parse it from the term_id in our ontology_parser, we don't take it as an input anymore

File: .github/workflows/push_tests.yml [hunk 3]
```diff
@@ -49,7 +49,7 @@ jobs:
             ${{ runner.os }}-pip-
       - name: Install dependencies
         run: |
-          pip install -r ./cellxgene_schema_cli/requirements.txt
+          pip install -r ./cellxgene_schema_cli/requirements.txt --extra-index-url https://test.pypi.org/simple/
```
[the remark this answers is not in the document]
Comment (nayib-jose-gloria) [reply .github/workflows/push_tests.yml:52]:
```suggestion
          pip install -r ./cellxgene_schema_cli/requirements.txt
```

**Review by Bento007 — APPROVED** [review]

**Pull request merged**